scripts/release.sh

```diff
@@ -314,10 +329,70 @@ main() {
     exit 1
   fi
 
-  # TODO: signing process
-  log_warning ""
-  log_warning "WARNING: The release has not been signed and published to github. This must be done manually."
-  log_warning ""
+  if [ "${DRY_RUN}" == "true" ] || [ "${NO_GH_RELEASE}" == 1 ]; then
+    log_warning ""
+    log_warning "WARNING: Skipping creating GitHub release, --no-gh-release is set."
+    log_warning "WARNING: If not running on DRY_MODE, please do the GitHub release manually."
+    log_warning ""
+  else
+    local gh_repo
+    local release_notes_temp_file
+    local release_url
+    local gh_release_args=()
+
+    # For the main branch (v3.6), we should mark the release as a prerelease.
+    # The release-3.5 (v3.5) branch, should be marked as latest. And release-3.4 (v3.4)
+    # should be left without any additional mark (therefore, it doesn't need a special argument).
+    if [ "${BRANCH}" = "main" ]; then
+      gh_release_args=(--prerelease)
+    elif [ "${BRANCH}" = "release-3.5" ]; then
```

Comment:
What happens when we release 3.6, this will not update? Maybe we should assert that branch name and fail that if new branch is cut we need to update which is the latest?